supernova fails to build with boost 1.67.0

Hey all,

I'm currently updating my PKGBUILD for the development version of SuperCollider for the [AUR](https://aur.archlinux.org).
However, supernova fails to build with current boost (1.67.0).

During boost rebuilds I was away, so another Arch Linux trusted user applied [this patch](https://git.archlinux.org/svntogit/community.git/tree/trunk/boost-1.67.patch?h=packages/supercollider), basically just casting the value to long.
If this is indeed a good fix for the problem at hand, let me know and I'll supply a pull request!

Environment
-----------

* SuperCollider version: latest `develop` (`g88621c5f91`)
* Operating system and version: Arch Linux

Expected Behavior
-----------------

supernova compiles

Current Behavior
----------------

supernova fails to build.


Steps to reproduce (for bugs)
-----------------------------

[Build SuperCollider ](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=supercollider-git) (here now with patch applied)

Error message (for bugs)
------------------------

```
Scanning dependencies of target libsupernova
[ 29%] Building CXX object server/supernova/CMakeFiles/libsupernova.dir/sc/sc_synth_definition.cpp.o
In file included from /build/supercollider-git/src/supercollider/server/supernova/sc/../server/synth.hpp:28,
                 from /build/supercollider-git/src/supercollider/server/supernova/sc/sc_synth.hpp:30,
                 from /build/supercollider-git/src/supercollider/server/supernova/sc/sc_synth_definition.cpp:25:
/build/supercollider-git/src/supercollider/server/supernova/./utilities/time_tag.hpp: In member function ‘boost::posix_time::ptime nova::time_tag::to_ptime() const’:
/build/supercollider-git/src/supercollider/server/supernova/./utilities/time_tag.hpp:232:102: error: no matching function for call to ‘boost::date_time::subsecond_duration<boost::posix_time::time_duration, 1000000>::subsecond_duration(double)’                                                                                                                                           
         time_duration offset = seconds(get_secs() - ntp_offset) + microseconds(get_nanoseconds()/1000);
                                                                                                      ^
In file included from /usr/include/boost/date_time/posix_time/posix_time_config.hpp:16,
                 from /usr/include/boost/date_time/posix_time/posix_time_system.hpp:13,
                 from /usr/include/boost/date_time/posix_time/ptime.hpp:12,
                 from /usr/include/boost/date_time/posix_time/posix_time_types.hpp:12,
                 from /usr/include/boost/thread/thread_time.hpp:11,
                 from /usr/include/boost/thread/lock_types.hpp:18,
                 from /usr/include/boost/thread/pthread/mutex.hpp:16,
                 from /usr/include/boost/thread/mutex.hpp:16,
                 from /usr/include/boost/thread/pthread/shared_mutex.hpp:14,
                 from /usr/include/boost/thread/shared_mutex.hpp:28,
                 from /build/supercollider-git/src/supercollider/external_libraries/nova-tt/nova-tt/rw_mutex.hpp:27,
                 from /build/supercollider-git/src/supercollider/server/supernova/./utilities/named_hash_entry.hpp:26,
                 from /build/supercollider-git/src/supercollider/server/supernova/sc/sc_synthdef.hpp:29,
                 from /build/supercollider-git/src/supercollider/server/supernova/sc/sc_synth_definition.hpp:23,
                 from /build/supercollider-git/src/supercollider/server/supernova/sc/sc_synth.hpp:28,
                 from /build/supercollider-git/src/supercollider/server/supernova/sc/sc_synth_definition.cpp:25:
/usr/include/boost/date_time/time_duration.hpp:285:14: note: candidate: ‘template<class T> boost::date_time::subsecond_duration<base_duration, frac_of_second>::subsecond_duration(const T&, typename boost::enable_if<boost::is_integral<T>, void>::type*)’                                                                                                                                  
     explicit subsecond_duration(T const& ss,
              ^~~~~~~~~~~~~~~~~~
/usr/include/boost/date_time/time_duration.hpp:285:14: note:   template argument deduction/substitution failed:
/usr/include/boost/date_time/time_duration.hpp: In substitution of ‘template<class T> boost::date_time::subsecond_duration<boost::posix_time::time_duration, 1000000>::subsecond_duration(const T&, typename boost::enable_if<boost::is_integral<T>, void>::type*) [with T = double]’:                                                                                                        
/build/supercollider-git/src/supercollider/server/supernova/./utilities/time_tag.hpp:232:102:   required from here
/usr/include/boost/date_time/time_duration.hpp:285:14: error: no type named ‘type’ in ‘struct boost::enable_if<boost::is_integral<double>, void>’
In file included from /usr/include/boost/date_time/posix_time/posix_time_config.hpp:16,
                 from /usr/include/boost/date_time/posix_time/posix_time_system.hpp:13,
                 from /usr/include/boost/date_time/posix_time/ptime.hpp:12,
                 from /usr/include/boost/date_time/posix_time/posix_time_types.hpp:12,
                 from /usr/include/boost/thread/thread_time.hpp:11,
                 from /usr/include/boost/thread/lock_types.hpp:18,
                 from /usr/include/boost/thread/pthread/mutex.hpp:16,
                 from /usr/include/boost/thread/mutex.hpp:16,
                 from /usr/include/boost/thread/pthread/shared_mutex.hpp:14,
                 from /usr/include/boost/thread/shared_mutex.hpp:28,
                 from /build/supercollider-git/src/supercollider/external_libraries/nova-tt/nova-tt/rw_mutex.hpp:27,
                 from /build/supercollider-git/src/supercollider/server/supernova/./utilities/named_hash_entry.hpp:26,
                 from /build/supercollider-git/src/supercollider/server/supernova/sc/sc_synthdef.hpp:29,
                 from /build/supercollider-git/src/supercollider/server/supernova/sc/sc_synth_definition.hpp:23,
                 from /build/supercollider-git/src/supercollider/server/supernova/sc/sc_synth.hpp:28,
                 from /build/supercollider-git/src/supercollider/server/supernova/sc/sc_synth_definition.cpp:25:
/usr/include/boost/date_time/time_duration.hpp:270:30: note: candidate: ‘boost::date_time::subsecond_duration<boost::posix_time::time_duration, 1000000>::subsecond_duration(const boost::date_time::subsecond_duration<boost::posix_time::time_duration, 1000000>&)’                                                                                                                         
   class BOOST_SYMBOL_VISIBLE subsecond_duration : public base_duration
                              ^~~~~~~~~~~~~~~~~~
/usr/include/boost/date_time/time_duration.hpp:270:30: note:   no known conversion for argument 1 from ‘double’ to ‘const boost::date_time::subsecond_duration<boost::posix_time::time_duration, 1000000>&’                                                                                                                                                                                   
/usr/include/boost/date_time/time_duration.hpp:270:30: note: candidate: ‘boost::date_time::subsecond_duration<boost::posix_time::time_duration, 1000000>::subsecond_duration(boost::date_time::subsecond_duration<boost::posix_time::time_duration, 1000000>&&)’                                                                                                                              
/usr/include/boost/date_time/time_duration.hpp:270:30: note:   no known conversion for argument 1 from ‘double’ to ‘boost::date_time::subsecond_duration<boost::posix_time::time_duration, 1000000>&&’                                                                                                                                                                                        
make[2]: *** [server/supernova/CMakeFiles/libsupernova.dir/build.make:63: server/supernova/CMakeFiles/libsupernova.dir/sc/sc_synth_definition.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:3338: server/supernova/CMakeFiles/libsupernova.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
```

--- a/server/supernova/utilities/time_tag.hpp
+++ b/server/supernova/utilities/time_tag.hpp
@@ -229,7 +229,7 @@ public:
 #ifdef BOOST_DATE_TIME_POSIX_TIME_STD_CONFIG
         time_duration offset = seconds(get_secs() - ntp_offset) + nanoseconds(get_nanoseconds());
 #else
-        time_duration offset = seconds(get_secs() - ntp_offset) + microseconds(get_nanoseconds()/1000);
+        time_duration offset = seconds(get_secs() - ntp_offset) + microseconds(static_cast<long>(get_nanoseconds()/1000));
 #endif
         return ptime(base, offset);
     }

